### ISSUE TITLE
Demo doesn't scroll DataTable in to view when focused

### ISSUE THREAD

**Issue by willmcgugan** [Textualize/textual#2771]
If you tab trough the widgets on `python -m textual`, the DataTable doesn't scroll in to view when it is focused. I'm certain it used to.

**Comment by darrenburns** [comment]
I can't reproduce this on `main`.

**Comment by willmcgugan** [comment]
Dave is seeing the same behaviour. You have to adjust to terminal size.

It doesn't actually appear to be data table related. Can occur in any control.

**Comment by darrenburns** [comment]
My `main` was outdated - I'm seeing it now. This means it must be a really recent thing.

What do you mean by "can occur in any control"? Do you mean there's other widgets impacted by this?

**Comment by darrenburns** [comment]
Ah ok - so it seems like the resizing is interfering in some way.

**Comment by willmcgugan** [comment]
You were right. I bisected (all by myself) to the following:

```
7049014faa79c423b8fa220731e65c04509882f0 is the first bad commit
commit 7049014faa79c423b8fa220731e65c04509882f0
Author: darrenburns <darrenburns@users.noreply.github.com>
Date:   Wed May 31 16:13:21 2023 +0100

    Option to ensure scroll_to_center doesn't scroll so as to hide the top left corner of the widget (#2682)

    * Option to ensure origin of widget is visible when calling scroll to center

    * Update CHANGELOG.md

    ---------

    Co-authored-by: Will McGugan <willmcgugan@gmail.com>

 CHANGELOG.md          |  1 +
 src/textual/screen.py |  2 +-
 src/textual/widget.py | 40 +++++++++++++++++++++++++++-------------
 3 files changed, 29 insertions(+), 14 deletions(-)
```

**Comment by willmcgugan** [comment]
Makes sense... https://github.com/Textualize/textual/commit/7049014faa79c423b8fa220731e65c04509882f0

**Comment by willmcgugan** [comment]
Fixed in 0.28.0
